index.rb

```diff
@@ -0,0 +1,23 @@
+def bubble_sort(array)
+  size = array.size
+  size.times do
+    array.each_with_index do |v, i|
+      break if i == size - 1
+
+      array[i], array[i + 1] = array[i + 1], array[i] if v > array[i + 1]
+    end
+  end
+  array
+end
+
+def bubble_sort_by(arr)
+  size = arr.size
+  size.times do
+    arr.each_with_index do |_, i|
+      break if i == size - 1
+
+      arr[i], arr[i + 1] = arr[i + 1], arr[i] if yield(arr[i], arr[i + 1]) >= 1
+    end
+  end
+  arr
+end
```

Comment:
-
 ```
 bubble_sort_by(["hi","hello","hey"]) do |left,right|
         left.length - right.length 
        end
```
#### returns  ‎👇
![Screenshot from 2020-01-16 15-53-26](https://user-images.githubusercontent.com/29695661/72562013-7053ea00-3878-11ea-8383-5bbb8ed29bc5.png)
#### instead of ‎👇
```# => ["hi", "hey", "hello"]```

Reply:
how do yo test it? because is working, maybe you forgot the p = in the image and it need to be

Reply:
` p = bubble_sort_by(["hi","hello","hey"]) do |left,right|
     left.length - right.length
  end`

Reply:
![image](https://user-images.githubusercontent.com/53204984/72563469-6085d700-3873-11ea-8ea8-3528270b58f1.png)

Reply:
![image](https://user-images.githubusercontent.com/53204984/72564251-f837f500-3874-11ea-8a81-efddd9744d64.png)